src/calendar_builder.ml

```diff
@@ -409,6 +431,28 @@ module Make_Precise(D: Date_sig.S)(T: Time_sig.S) = struct
 
   (*S Period. *)
 
+  let int_of_month = function
+    | Jan -> 1 | Feb -> 2 | Mar -> 3
+    | Apr -> 4 | May -> 5 | Jun -> 6
+    | Jul -> 7 | Aug -> 8 | Sep -> 9
+    | Oct -> 10 | Nov -> 11 | Dec -> 12
+
+  let has_expired a b y m d =
```

Comment:
`y m d` should be a single parameter no?
Can you find better names for `a` and `b` possibly turn them into named parameters?

Reply:
For `a` and `b`, no problem.

But for the type `Period.t` it's not what you think it is, we need 3 int here. I will try to make something prettier though.

Reply:
T'es sûr ? `src/date_sig.mli` :

```
(** A period is the number of days between two dates. *)
module Period : sig
  ...
  val make: int -> int -> int -> t
```

Reply:
```ocaml
type t = field date
type -'a date constraint 'a = [< field ]
type field = Period.date_field
type date_field = [ `Day | `Month | `Week | `Year ]
```

I don't think that's what we want here.

Reply:
Otherwise, I have made some changes you can look at.